### PR TITLE
Add default TopologySpreadContraints to distributor

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**
Issue: https://issues.redhat.com/browse/LOG-3850

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/internal/manifests/var.go [hunk 1]
```diff
@@ -135,6 +135,16 @@ func commonLabels(stackName string) map[string]string {
 	}
 }
 
+func componentInstaceLabels(l labels.Set) map[string]string {
+	// This code assumes that this function will never be called with a set of labels
+	// that don't have the "component" and "instance" labels since we enforce those on
+	// all the components of the LokiStack
+	return map[string]string{
+		kubernetesInstanceLabel:  l[kubernetesInstanceLabel],
+		kubernetesComponentLabel: l[kubernetesComponentLabel],
```
Comment (aminesnow) [inline operator/internal/manifests/var.go:144]: It's much better to have the signature be very clear than use a map for parameters.
```suggestion
func componentInstaceLabels(component string, stackName string) map[string]string {
	return map[string]string{
	        kubernetesInstanceLabel:  stackName,
		kubernetesCompomentLabel: component,
					
```

File: operator/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ## Main
 
 - [9366](https://github.com/grafana/loki/pull/9366) **periklis**: Add support for custom tenant topology in rules
+- [9383](https://github.com/grafana/loki/pull/9383) **JoaoBraveCoding**: Add default TopologySpreadContaints to Distributor
```
Comment (aminesnow) [inline operator/CHANGELOG.md:4]:
```suggestion
- [9383](https://github.com/grafana/loki/pull/9383) **JoaoBraveCoding**: Add default TopologySpreadContaints to Distributor
- [9366](https://github.com/grafana/loki/pull/9366) **periklis**: Add support for custom tenant topology in rules
```

**Review by aminesnow — CHANGES_REQUESTED** [review]
One change request, otherwise LGTM!

File: operator/internal/manifests/distributor.go [hunk 3]
```diff
@@ -59,7 +59,8 @@ func NewDistributorDeployment(opts Options) *appsv1.Deployment {
 	l := ComponentLabels(LabelDistributorComponent, opts.Name)
 	a := commonAnnotations(opts.ConfigSHA1, opts.CertRotationRequiredAt)
 	podSpec := corev1.PodSpec{
-		Affinity: configureAffinity(l, opts.Gates.DefaultNodeAffinity),
+		Affinity:                  configureAffinity(l, opts.Gates.DefaultNodeAffinity),
+		TopologySpreadConstraints: defaultTopologySpreadConstraints(l),
```
Comment (periklis) [inline operator/internal/manifests/distributor.go:63]: This definition is overwritten by and thus no node-spreading is happening when zone-aware data replication is on. See:
https://github.com/grafana/loki/blob/520e434446e4c646ecc2678f3c3b57fff69643cc/operator/internal/manifests/var.go#L146-L166

**Review by periklis — CHANGES_REQUESTED** [review]

File: operator/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ## Main
 
 - [9406](https://github.com/grafana/loki/pull/9406) **aminesnow**: Add label selector to zone awareness TopologySpreadConstraints
+- [9383](https://github.com/grafana/loki/pull/9383) **JoaoBraveCoding**: Add default TopologySpreadContaints to Distributor
```
Comment (periklis) [inline operator/CHANGELOG.md:4]:
```suggestion
- [9383](https://github.com/grafana/loki/pull/9383) **JoaoBraveCoding**: Add default TopologySpreadContaints to Distributor
- [9406](https://github.com/grafana/loki/pull/9406) **aminesnow**: Add label selector to zone awareness TopologySpreadConstraints
```

File: operator/internal/manifests/var.go [hunk 5]
```diff
@@ -154,7 +164,7 @@ func topologySpreadConstraints(spec lokiv1.ReplicationSpec, component string, st
 				WhenUnsatisfiable: corev1.DoNotSchedule,
 				LabelSelector: &metav1.LabelSelector{
 					MatchLabels: map[string]string{
-						kubernetesCompomentLabel: component,
+						kubernetesComponinternal/manifests/distributor_test.goentLabel: component,
```
Comment (periklis) [inline operator/internal/manifests/var.go:167]: Edit typo?

**Review by periklis — CHANGES_REQUESTED** [review]

**Review by periklis — APPROVED** [review]

**Pull request merged**